src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx

```diff
@@ -63,19 +68,57 @@ const CreateOrEditQuizContainer = ({
     })
   }
 
-  const createAddQuizService = useCallback(
+  const createQuizService = useCallback(
     (data?: CreateQuizParams) => ResourceService.addQuiz(data),
     []
   )
 
   const { fetchData: addNewQuiz } = useAxios<Quiz, CreateQuizParams>({
-    service: createAddQuizService,
+    service: createQuizService,
     fetchOnMount: false,
     defaultResponse,
     onResponse: handleResponse,
-    onResponseError: onResponseError
+    onResponseError
+  })
+
+  const editQuizService = useCallback(
+    (params?: UpdateQuizParams) => ResourceService.editQuiz(params),
+    []
+  )
+
+  const { fetchData: fetchEditedQuiz } = useAxios<null, UpdateQuizParams>({
+    service: editQuizService,
+    fetchOnMount: false,
+    defaultResponse: null,
+    onResponse: handleResponse,
+    onResponseError
+  })
+
+  const getQuiz = (id?: string): Promise<AxiosResponse> => {
+    return ResourceService.getQuiz(id)
+  }
+
+  const handleGetQuizResponse = (quiz: Quiz) => {
+    setTitle(quiz.title)
+    setDescription(quiz.description)
+    setQuestions(quiz.items)
+  }
+
+  const { fetchData: fetchQuizData } = useAxios<Quiz, string>({
+    service: getQuiz,
+    fetchOnMount: false,
+    defaultResponse,
+    onResponse: handleGetQuizResponse,
+    onResponseError
   })
 
+  useEffect(() => {
+    if (id) {
+      void fetchQuizData(id)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id])
```

Comment:
if you add fetchQuizData to the dependency array it will make some extra renders?

Reply:
Yes, it will be the same story that we had on the lesson page.